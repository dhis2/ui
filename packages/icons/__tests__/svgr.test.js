import path from 'path'
import fs from 'fs'
import rimraf from 'rimraf'
import execa from 'execa'

describe('svg build', () => {
    it('should generate the expected component', async () => {
        const packageRoot = path.join(__dirname, '..')
        const testPath = path.join(packageRoot, 'test')
        const componentPath = path.join(testPath, 'Add16.js')

        // First remove the target dir to ensure we're not testing against an older file
        rimraf.sync(testPath)
        expect(fs.existsSync(componentPath)).toBe(false)

        // Generate the component with svgr-cli
        await execa.command('npx svgr -d test src/add-16.svg', {
            // svgr-cli locates files relative to process.cwd() so we need to set the cwd explicitly
            cwd: packageRoot,
        })

        // Assert that the file has been generated and that it matches our expectations
        expect(fs.existsSync(componentPath)).toBe(true)
        const componentBuffer = fs.readFileSync(componentPath)
        expect(componentBuffer.toString()).toMatchSnapshot()
    })
})
