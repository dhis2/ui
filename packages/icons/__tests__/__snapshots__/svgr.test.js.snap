// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`svg build should generate the expected component 1`] = `
"import * as React from 'react'
import propTypes from '@dhis2/prop-types'

function SvgAdd16({ color }) {
    return (
        <svg
            height={16}
            viewBox=\\"0 0 16 16\\"
            width={16}
            xmlns=\\"http://www.w3.org/2000/svg\\"
            color={color}
        >
            <path
                d=\\"M8 2a1 1 0 011 1v4h4a1 1 0 010 2H9v4a1 1 0 01-2 0V9H3a1 1 0 110-2h4V3a1 1 0 011-1z\\"
                fill=\\"currentColor\\"
                fillRule=\\"evenodd\\"
            />
        </svg>
    )
}

SvgAdd16.propTypes = {
    color: propTypes.string,
}
export default SvgAdd16
"
`;
